### DISCUSSION TITLE
Request new version of JBSIM for Unreal 5.4/5.5

### DISCUSSION THREAD

**Discussion by steamjunkie** [JSBSim-Team/jsbsim#1205] [General]
Request new version of JBSIM for Unreal 5.4/5.5

**Comment by bcoconni** [comment]
I'm not sure what you mean by that ? Are you getting errors trying to build the JSBSim component for UE 5.4 and 5.5 ?
According to discussion #1180, I understood that JSBSim should build on UE 5.4. @gallonmate, is that correct ?

**Reply by steamjunkie** [reply]
I not been able to build version 2 of jbsim on unreal 5.4 or5.5 so far i get lots of build compilation errors. What version compiles in vs22?.  0r What version of compiler and what vs lib do i need.  I will try it again.

**Reply by steamjunkie** [reply]
OK I got it build on 5.5.1, my issue was corrupt source for what ever reason.  so working now.  thanks for feedback.

**Comment by gallonmate** [comment]
I just tested UE5.4. I will submit a pull request within the next few days/week:

Here's the solution:
First UE5.4 needs it's users to install `MSVC v143 x64/x86 build tools (v14.38-17.8)`
This is done through Visual Studio Installer, open it and then click modify on the Install. Next click the individual components tab at the top. Search for _MSVC v143 x64/x86 build tools (v14.38-17.8)_ and install it.

Next, two files need a small edit found in your project folder `UnrealEngine\Source`:
`UEReferenceApp.Target.cs`
`UEReferenceAppEditor.Target.cs`

Open both files and edit this line `DefaultBuildSettings = BuildSettingsVersion.V2` 
Change  `V2` to `Latest`.
Example: `DefaultBuildSettings = BuildSettingsVersion.Latest`

Now UE5.4 JSBSim should compile. 
UE5.5 should also work, according to another member using similar steps https://github.com/JSBSim-Team/jsbsim/discussions/1180#discussioncomment-11651838

I will test UE5.5 soon to confirm and check backwards compatibility. Then I will submit a pull request, along with updating the Read Me file.

**Reply by bcoconni** [reply]
Thanks for your support, @gallonmate . Just for my information, will these modifications allow using any version of UE5 ? Or will they break backward compatibility with UE <= 5.3 ?

**Reply by gallonmate** [reply]
Yes these changes allow for all prior versions, and future versions! Using the `Latest` build setting, the engine version will select the correct build version per engine.

@bcoconni Thanks to you as well! :christmas_tree: :santa:
